test/diags_test.py

```diff
@@ -106,7 +106,10 @@ def execute(self, test_str, imagefilename, imagethreshold, ncfiles, rtol, atol):
         proc = subprocess.Popen([self.diagstr], shell=True)
         proc_status = proc.wait()
         if proc_status!=0: 
-            raise DiagError("diags run failed")
+            proc = subprocess.Popen([self.diagstr], shell=True)
+            proc_status = proc.wait()
+            if proc_status!=0: 
+                raise DiagError("diags run failed")
```

Comment:
@jccosta I think the `random` failures are due to the template. @painter1 I will merge this in, but @jccosta please do not forget to remove this when you fix the "random" template failure.